presto-spi/src/main/java/io/prestosql/spi/block/MapBlock.java

```diff
@@ -294,4 +280,13 @@ public Block getLoadedBlock()
                 loadedValueBlock,
                 hashTables);
     }
+
+    @Override
+    protected void ensureHashTableLoaded()
+    {
+        // this is double checked locking
+        if (!hashTables.isBuilt()) {
```

Comment:
Instead of checking here, you could add a `buildAllHashTablesIfNecessary` and encapsulate the check there. That way, callers don't have to reason about it atomicity and thread-safety.